python/ray/serve/_private/http_state.py

```diff
@@ -49,6 +50,10 @@ def __init__(
         self._proxy_actors: Dict[NodeId, ActorHandle] = dict()
         self._proxy_actor_names: Dict[NodeId, str] = dict()
         self._head_node_id: str = head_node_id
+
+        # Defer the initialization only when it's used.
+        self._gcs_client = None
```

Comment:
why delay this?

Reply:
There is one test (https://github.com/ray-project/ray/blob/HEAD/python/ray/serve/tests/test_http_state.py) running without ray, so the connection will fail.

Do you think I should just make it run with ray?

Reply:
Ah I see, can we pass the gcs_client into `HTTPState` and in this unit test pass in a mock?